server/cache.go

```diff
@@ -531,6 +531,10 @@ func (c *clusterInfo) handleRegionHeartbeat(region *regionInfo) error {
 		return c.putRegionLocked(region)
 	}
 
+	if region.Leader.GetId() != origin.Leader.GetId() {
+		log.Infof("update region leader %v origin %v", region.Leader, origin.Leader)
```

Comment:
Need to log region id, or I think you may just log `region` and `origin` as above.